storey/aggregations.py

```diff
@@ -17,23 +19,19 @@ class AggregateByKey(Flow):
     Persistence is done via the `WriteToTable` step and based on the Cache object persistence settings.
 
     :param aggregates: List of aggregates to apply for each event.
-    :type aggregates: list of FieldAggregator
     :param table: A Table object to aggregate the data into.
-    :type table: Table
     :param key: Key field to aggregate by, accepts either a string representing the key field or a key extracting function.
      Defaults to the key in the event's metadata. (Optional)
-    :type key: string or Function (Event=>object)
     :param emit_policy: Policy indicating when the data will be emitted. Defaults to EmitEveryEvent. (Optional)
-    :type emit_policy: {EmitEveryEvent, EmitAfterMaxEvent, EmitAfterPeriod, EmitAfterWindow}
     :param augmentation_fn: Function that augments the features into the event's body. Defaults to updating a dict. (Optional)
-    :type augmentation_fn: Function ((Event, dict) => Event)
     :param enrich_with: List of attributes names from the associated storage object to be fetched and added to every event. (Optional)
-    :type enrich_with: list of string
     :param aliases: Dictionary specifying aliases to the enriched columns, of the format `{'col_name': 'new_col_name'}`. (Optional)
-    :type aliases: dict
     """
 
-    def __init__(self, aggregates, table, key=None, emit_policy=_default_emit_policy, augmentation_fn=None, enrich_with=None, aliases=None,
+    def __init__(self, aggregates: Union[List[FieldAggregator], List[dict]], table: Table, key: Union[str, Callable, None] = None,
+                 emit_policy: Union[EmitEveryEvent, FixedWindows, SlidingWindows, EmitAfterPeriod, EmitAfterWindow,
+                                    EmitAfterMaxEvent, dict] = _default_emit_policy,
```

Comment:
I think at this point it might be justified to define a common class, even if it's empty?

Reply:
Discussed offline and decided not to, at the moment